client/replicate/common/strings.go

```diff
@@ -67,3 +67,32 @@ func MatchStrictRegex(pattern, str string) (bool, error) {
 func CompileStrictRegex(pattern string) (*regexp.Regexp, error) {
 	return regexp.Compile(BuildStrictRegex(pattern))
 }
+
+func PrepareTLD(namespace, tld string) string {
+	subdomains := strings.Split(tld, ".")
+	subdomains[0] = namespace
+
+	return strings.Join(subdomains, ".")
+}
+
+func PrepareRouteMatch(namespace, match string) string {
+	if !strings.Contains(match, "Host") {
+		return match
+	}
+
+	parts := strings.Split(match, "`")
```

Comment:
This will work only for parts that have a single back-tick quoted host. However, each parenthesis-contained filter can include a list of hosts/paths. It would be better to split on `(` and `)` and then parse each string inside.

Example of this here: https://go.dev/play/p/U0pnE6dCb4t